stdlib/public/Concurrency/TaskGroup+addTask.swift.gyb

```diff
@@ -0,0 +1,408 @@
+      % # In this case, we already added the pending task count before we create the task
+      % # so we must not add to the pending counter again.
+      addPendingGroupTaskUnconditionally: false,
+      % else:
+      addPendingGroupTaskUnconditionally: true,
+      % end
+      isDiscardingTask: ${str(IS_DISCARDING).lower()}
+    )
+
+    let builtinSerialExecutor =
+      Builtin.extractFunctionIsolation(operation)?.unownedExecutor.executor
+
+    var task: Builtin.NativeObject?
+
+    % # These #if conditionals are in specific order because if we have the "latest one",
+    % # we definitely have the prior ones, as we always keep adding parameters.
+    % if HAS_TASK_NAME:
+    #if $BuiltinCreateAsyncTaskName
+    if var name {
+      task =
+        name.withUTF8 { nameBytes in
+          Builtin.createTask(
+            flags: flags,
+            initialSerialExecutor: builtinSerialExecutor,
+            taskGroup: _group,
+            initialTaskExecutorConsuming: taskExecutor,
+            taskName: nameBytes.baseAddress?._rawValue,
+            operation: operation).0
+        }
+    }
+    #endif // $BuiltinCreateAsyncTaskName
+    % end # HAS_TASK_NAME
+
+    // Task name was not set, or task name createTask is unavailable
+    % if HAS_TASK_EXECUTOR:
+    if task == nil, let taskExecutor {
+      #if $BuiltinCreateAsyncTaskOwnedTaskExecutor
+      task = Builtin.createTask(
+        flags: flags,
+        initialSerialExecutor: builtinSerialExecutor,
+        taskGroup: _group,
+        initialTaskExecutorConsuming: taskExecutor,
+        operation: operation).0
+      #else
+      // legacy branch for the non-consuming task executor
+      let executorBuiltin: Builtin.Executor =
+        taskExecutor.asUnownedTaskExecutor().executor
+
+      task = Builtin.createTask(
+        flags: flags,
+        initialSerialExecutor: builtinSerialExecutor,
+        taskGroup: _group,
+        initialTaskExecutor: executorBuiltin,
+        operation: operation).0
+      #endif
+    }
+    % end
+
+    % # The baseline fallback, if no other create calls could be used
+    if task == nil {
+      task = Builtin.createTask(
+        flags: flags,
+        initialSerialExecutor: builtinSerialExecutor,
+        taskGroup: _group,
+        operation: operation).0
+    }
+
+    // Assert that we did create the task, but there's no need to store it,
+    // as it was added to the group itself.
+    assert(task != nil, "Expected task to be created!")
+    % if IS_ADD_UNLESS_CANCELLED:
+    return true // task successfully enqueued
+    % end
+    % end # !IS_IMPL_UNAVAILABLE
+  }
+
+}
+% if IFDEF:
+#endif
+% end
+
+% end
+% end
+% end
+
+// FIXME: The following sources should be in SourceCompatibilityShims.swift,
+//        but the current build has trouble including the generated gyb sources
+//        such that they are available to the same module. As we move to the
+//        new build style we should resolve this. rdar://145171772
```

Comment:
Workaround since otherwise the SourceCompatibilityShims.swift file would not compile, as it would not see the gyb generated add methods